src/ips.rs

```diff
@@ -273,28 +272,28 @@ impl Ips {
         nodes: &[Node],
         ratings: &mut [PeerEntry],
     ) {
-        let selected_location = &selected_node
-            .geolocation
-            .as_ref()
-            .map(|geo_info| {
-                Location::new(
-                    geo_info.latitude.unwrap_or_default(), // TODO(asmie): refactor when zg-core will be updated with Location
-                    geo_info.longitude.unwrap_or_default(),
-                )
-            })
-            .unwrap_or(Location::new(0.0, 0.0));
+        if selected_node.geolocation.is_none() {
+            return;
+        }
+
+        let selected_location =
+            if let Some(location) = selected_node.geolocation.as_ref().unwrap().location {
```

Comment:
but I've been thinking, it feels a bit funny to have something named `location` unwrapped twice.
long term alternative could be:

location
-> city/country name
-> can have other stuff
-> has coordinates

so the `location` it can be a broader term - like our outer struct

while the term coordinates are mainly defined by the longitude and latitude
[Latitude and Longitude are the units that represent the coordinates at geographic coordinate system. ](https://www.latlong.net/)

coordinate
-> longitude
-> latitude

Reply:
I have to say, I didn't see this before, but I kind of agree. `coordinates` (and `Coordinates`) does seem intuitively better.

Reply:
Well, that's no problem to refactor that name.